### PR TITLE
Execute template through django-admin startproject

### PR DESCRIPTION
**Pull request by davish**
This PR makes this an actual django template in the eyes of Django (https://docs.djangoproject.com/en/2.2/ref/django-admin/#startproject). Instructions to start a new project from it are available in the `README`.

@pawalt wanna make sure I didn't break helm... not sure the best way to test that
[ch116]

### CODE REVIEW

**Review by pawalt — DISMISSED** [review]
LGTM. I dont think it'll break helm. Will find out soon. I'll have to change labs-helm to use `values.yml` though.

**Review by pawalt — DISMISSED** [review]
Still fine

File: .circleci/config.yml
Comment (ArmaanT) [inline .circleci/config.yml]: I don't think we'll always want the docker image name to be the same as the project name. For example with clubs, the backend image is called `clubs-backend`, but the project name is just `pennclubs`

File: k8s/chart/values.yml
Comment (ArmaanT) [inline k8s/chart/values.yml]: Shouldn't this also be replaced by startproject?

File: k8s/values.yml
Comment (ArmaanT) [inline k8s/values.yml]: Same issue with the docker image. Not sure we want this to always be the same as project name.

File: manage.py
Comment (ArmaanT) [inline manage.py]: dev -> development

File: setup.cfg
Comment (ArmaanT) [inline setup.cfg:8]: We should still indicate that more than just the project name will need to be put here. Like `courses`, `plan`, etc in PCX

File: project_name/settings/base.py
Comment (ArmaanT) [inline project_name/settings/base.py]: Should this refer to `BACKEND_DOMAIN`?

File: project_name/settings/staging.py
Comment (ArmaanT) [inline project_name/settings/staging.py]: Not a huge fan of  f-strings since they can execute arbitrary python code.

File: README.md
Comment (ArmaanT) [inline README.md:69]: I think we should still keep some of the k8s instructions because we aren't auto setting all of those.

**Review by ArmaanT — CHANGES_REQUESTED** [review]

File: .circleci/config.yml
Comment (davish) [reply .circleci/config.yml]: does it necessarily matter for new projects? this could start a convention of keeping image names the same as project names

File: k8s/chart/values.yml
Comment (davish) [reply k8s/chart/values.yml]: @pawalt  said that the `values.yml` in the chart is global defaults for everything, but i'd defer to him on this one.

File: k8s/values.yml
Comment (davish) [reply k8s/values.yml]: similar to comment above, don't know if it's a bad thing for simplicity's sake to start a new convention

File: manage.py
Comment (davish) [reply manage.py]: done

File: setup.cfg
Comment (davish) [reply setup.cfg:8]: i think that can be specified in the README, considering that when `startproject` is called there's no first party apps, would be weird if we make people put in more modules to lint it that might not be created yet.

File: project_name/settings/base.py
Comment (davish) [reply project_name/settings/base.py]: done

File: project_name/settings/staging.py
Comment (davish) [reply project_name/settings/staging.py]: oh for real? yike

File: README.md
Comment (davish) [reply README.md:69]: deferring again to @pawalt here

File: .circleci/config.yml
Comment (pawalt) [reply .circleci/config.yml]: This is the template-django project, not the unified project. Unified project will have separated image names.

File: k8s/chart/values.yml
Comment (pawalt) [reply k8s/chart/values.yml]: Davis is correct on this one. Although, we should probably change this to example.org so that it doesn't conflict with the currently-active `pennlabs.org` deployment.

File: k8s/values.yml
Comment (pawalt) [reply k8s/values.yml]: I think this is a good convention to have.

File: project_name/settings/staging.py
Comment (pawalt) [reply project_name/settings/staging.py]: f-strings can't change arbitrary python code if the thing you pass into them is a string. They can only execute arbitrary python code if you said something like `f'https://{print(hi)}`. Since `BACKEND_DOMAIN` is a string, we don't need to be worried about this. We should be using f-strings here.

File: README.md
Comment (pawalt) [reply README.md:69]: I think k8s docs don't belong here. These settings are going to be pretty similar across products, so i'd rather make a general guide in Notion.

File: project_name/settings/base.py
Comment (pawalt) [inline project_name/settings/base.py]: Let's use f-strings here

**Review by pawalt — CHANGES_REQUESTED** [review]

File: .circleci/config.yml
Comment (ArmaanT) [reply .circleci/config.yml]: We currently have CORS/CSRF configuration in this repo that only applies to unified projects. We should probably make a decision on how we want to split things up. I think it probably makes sense to have all (backend only and unified) django configuration in this repo, but then just have a note to delete certain things for backend only. The k8s probably should be specific to backend only, since we need to provide a new chart for a unified product anyway.

File: k8s/chart/values.yml
Comment (ArmaanT) [reply k8s/chart/values.yml]: This will just depend on the result on the thread above.

File: k8s/values.yml
Comment (ArmaanT) [reply k8s/values.yml]: Same here.

File: setup.cfg
Comment (ArmaanT) [reply setup.cfg:8]: Putting it somewhere in the README makes sense to me.

File: project_name/settings/staging.py
Comment (ArmaanT) [reply project_name/settings/staging.py]: Agreed. After testing, it doesn't appear there's a way to execute code through an f-string.

File: .circleci/config.yml
Comment (davish) [reply .circleci/config.yml]: yeah i agree that there is normally an associated frontend with a backend. don't know if that changes helm stuff or timelines @pawalt

File: README.md
Comment (ArmaanT) [reply README.md:69]: I think it may make more sense to create a script to do all the replacing for a new project and just include that instead.

File: project_name/settings/base.py
Comment (ArmaanT) [reply project_name/settings/base.py]: I also just realized, we have a weird mix of where we're defining settings. For example the db is defined in `base.py` with a fallback to dev settings, but then we have things like the backend/frontend_domain variables which are in `base.py` but default to dummy entries. We should probably try to standardize this.

File: project_name/settings/base.py
Comment (ArmaanT) [inline project_name/settings/base.py]: Should we keep this as the literal string secret_key to ensure people remember to set a proper secret key as an environment variable?

Comment (davish) [reply project_name/settings/base.py]: I think django sets a default secret key for a reason -- want apps to work in development right away. Not opposed to putting `SECRET_KEY = os.environ.get('SECRET_KEY', None)` as an override in `settings.production`, though.

Comment (ArmaanT) [reply project_name/settings/base.py]: I think django generates one because it doesn't assume the secret key will be uploaded to github. For closed-source applications, that makes the most sense. Setting the default (and by proxy development) secret key to just 'secret_key' would still allow django to work in development.

**Comment by ArmaanT** [comment]
I haven't actually tested any of the above changes. We definitely need to before this PR is merged.
[ch298]
[ch212]
[ch85]
[ch306]
[ch307]
[ch91]
[ch39]
[ch309]

File: Pipfile
Comment (pawalt) [inline Pipfile:10]: if we're using black, do we still need this?

File: setup.cfg
Comment (pawalt) [inline setup.cfg:2]: ditto on black comment

**Review by pawalt — CHANGES_REQUESTED** [review]

File: project_name/settings/development.py
Comment (davish) [inline project_name/settings/development.py]: What about in `ci.py`? will `PLATFORM_ACCOUNTS` not being defined throw any errors?

File: Pipfile
Comment (ArmaanT) [reply Pipfile:10]: Yep, there are still a couple of things that black lets through that we wouldn't want. Import order is one of them (and fixed using isort)

File: Pipfile
Comment (davish) [inline Pipfile]: did you add CORS to the list of middlewares too?

**Comment by davish** [comment]
don't think i can review because this is my PR, but this looks pretty good. ahoy

File: Pipfile
Comment (ArmaanT) [reply Pipfile]: Thanks, missed that! Fixed

File: project_name/settings/development.py
Comment (ArmaanT) [reply project_name/settings/development.py]: `ci.py` will use the values defined in `base.py`, but in CI responses are all mocked when testing, so it shouldn't matter.

**Review by ArmaanT — CHANGES_REQUESTED** [review]
LGTM! We should just change `yml`->`yaml`

**Review by ArmaanT — APPROVED** [review]
LGTM

**Pull request merged**
